contracts/BananaAllocator.sol

```diff
@@ -0,0 +1,235 @@
+pragma solidity 0.6.12;
+
+/*
+ * ApeSwapFinance
+ * App:             https://apeswap.finance
+ * Medium:          https://medium.com/@ape_swap
+ * Twitter:         https://twitter.com/ape_swap
+ * Telegram:        https://t.me/ape_swap
+ * Announcements:   https://t.me/ape_swap_news
+ * GitHub:          https://github.com/ApeSwapFinance
+ */
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+// Stubbed import to get price for allocation token
+import "./PriceGetter.sol";
+import "./interfaces/IMasterApe.sol";
+
+// TODO: Write tests 
+
+contract BananaAllocator is Ownable {
+    using SafeMath for uint256;
+
+    IERC20 public allocationToken; // BANANA token
+    PriceGetter public priceGetter; // Stubbed way to grab price
+    IMasterApe public masterApe; // Stubbed way to grab price
+
+    // Information for each allocation
+    struct AllocationInfo {
+        string allocationName; // For recordkeeping purposes
+        uint256 allocationAmount; // Amount (intended to be percentages, but can be treated similarly to farm multipliers)
+        uint256 tokensAvailable; // The current amount of tokens available to a specific allocation
+        mapping(address => bool) allocationAdmins; // People approved to extract tokens allocated to a specific allocation
```

Comment:
renaming to `isAllocationAdmin` would help with readability. `allocationAdmins` to me sounds like an array of admins.